### DISCUSSION TITLE
How can docket_id and person_id be joined?

### DISCUSSION THREAD

**Discussion by eshanprashar** [freelawproject/courtlistener#4176] [General]
Hi @mlissner! I wanted to conduct an analysis on judges and types of cases they've worked on. My plan was to write an api request that could fetch docket_id and the person_id. In the data model, this seems to be available in the table **search_docket_panel** but I couldn't find an API corresponding to that. I'd appreciate any guidance on how to proceed for this request!

**Comment by mlissner** [comment]
The judge fields can be complicated. Are you thinking about federal cases in our PACER database or case law?

**Reply by eshanprashar** [reply]
I am not confident about the difference between the two - is PACER a subset of case law? Assuming I were interested in the latter, how could I proceed?

**Reply by mlissner** [reply]
Case law is decisions and opinions and PACER data is anything from a federal case. 

If you want to focus on case law, I'd take a deeper look at the data model and see what you can learn that way. We don't have perfect judge information, in the case law DB, but I'd start by finding all the related fields and see which ones make sense to look at.

**Reply by eshanprashar** [reply]
Does that mean PACER is the superset and case law is a subset since it contains only decisions and opinions? When I try to access the recap api, I get this error: 
![image](https://github.com/freelawproject/courtlistener/assets/114944808/7baa7ba0-e211-47d9-a80a-507fd0240ab5)

I got the same error while exploring the Attorney API:
![image](https://github.com/freelawproject/courtlistener/assets/114944808/f01aa835-5d32-4239-9a23-be3ab9028dd3)

I also noticed that the opinions object has a key called **author** - does this refer to the judge who gave the opinion?

**Reply by eshanprashar** [reply]
Does the **assigned_to** field in the docket API supposed to have the name of the judge? Sorry for the spam!

**Comment by eshanprashar** [comment]
Hey @mlissner : is it possible to get any guidance on the 2 things above:
* How can a user resolve api access errors like the one shown for **recap api**?
* Is the assigned_to field in the **docket API** supposed to have the name of the judge?

**Reply by mlissner** [reply]
The RECAP/PACER APIs are not open. Send an email to me and we can discuss: mike@free.law.

The assigned to field is generally populated for dockets in the PACER data set, yes.
